Mathlib/CategoryTheory/Sites/RegularExtensive.lean

```diff
@@ -368,4 +377,87 @@ lemma isSheaf_of_projective (F : Cᵒᵖ ⥤ Type (max u v)) [Preregular C] [∀
 
 end RegularSheaves
 
+section ExtensiveSheaves
+
+variable [FinitaryPreExtensive C] {C}
+
+/-- A presieve is *extensive* if it is finite and its arrows induce an isomorphism from the
+coproduct to the target. -/
+class Presieve.extensive [HasFiniteCoproducts C] {X : C} (R : Presieve X) :
+    Prop where
+  /-- `R` consists of a finite collection of arrows that together induce an isomorphism from the
+  coproduct of their sources. -/
+  arrows_sigma_desc_iso : ∃ (α : Type) (_ : Fintype α) (Z : α → C) (π : (a : α) → (Z a ⟶ X)),
+    R = Presieve.ofArrows Z π ∧ IsIso (Sigma.desc π)
```

Comment:
The `IsIso` condition should be stated in terms of a colimit cofan. This should also allow dropping the assumption `[HasFiniteCoproducts C]`. This may require some changes in the code below, but otherwise the PR looks good to me.

Reply:
Done. I also added a todo to use general colimit API in the definition of extensive coverage, but I'll do this in a separate PR as that requires some refactoring in other files, such as the effective epi one.